sentry/src/main/java/io/sentry/SentryUUID.java

```diff
@@ -0,0 +1,299 @@
+    final char[] uuidChars = new char[SENTRY_SPAN_UUID_STRING_LENGTH];
+
+    fillMostSignificantBits(uuidChars, mostSignificantBits);
+
+    return new String(uuidChars);
+  }
+
+  private static void fillMostSignificantBits(
+      final char[] uuidChars, final long mostSignificantBits) {
+    uuidChars[0] = HEX_DIGITS[(int) ((mostSignificantBits & 0xf000000000000000L) >>> 60)];
+    uuidChars[1] = HEX_DIGITS[(int) ((mostSignificantBits & 0x0f00000000000000L) >>> 56)];
+    uuidChars[2] = HEX_DIGITS[(int) ((mostSignificantBits & 0x00f0000000000000L) >>> 52)];
+    uuidChars[3] = HEX_DIGITS[(int) ((mostSignificantBits & 0x000f000000000000L) >>> 48)];
+    uuidChars[4] = HEX_DIGITS[(int) ((mostSignificantBits & 0x0000f00000000000L) >>> 44)];
+    uuidChars[5] = HEX_DIGITS[(int) ((mostSignificantBits & 0x00000f0000000000L) >>> 40)];
+    uuidChars[6] = HEX_DIGITS[(int) ((mostSignificantBits & 0x000000f000000000L) >>> 36)];
+    uuidChars[7] = HEX_DIGITS[(int) ((mostSignificantBits & 0x0000000f00000000L) >>> 32)];
+    uuidChars[8] = HEX_DIGITS[(int) ((mostSignificantBits & 0x00000000f0000000L) >>> 28)];
+    uuidChars[9] = HEX_DIGITS[(int) ((mostSignificantBits & 0x000000000f000000L) >>> 24)];
+    uuidChars[10] = HEX_DIGITS[(int) ((mostSignificantBits & 0x0000000000f00000L) >>> 20)];
+    uuidChars[11] = HEX_DIGITS[(int) ((mostSignificantBits & 0x00000000000f0000L) >>> 16)];
+    uuidChars[12] = HEX_DIGITS[(int) ((mostSignificantBits & 0x000000000000f000L) >>> 12)];
+    uuidChars[13] = HEX_DIGITS[(int) ((mostSignificantBits & 0x0000000000000f00L) >>> 8)];
+    uuidChars[14] = HEX_DIGITS[(int) ((mostSignificantBits & 0x00000000000000f0L) >>> 4)];
+    uuidChars[15] = HEX_DIGITS[(int) (mostSignificantBits & 0x000000000000000fL)];
+  }
+
+  static long getHexValueForChar(final char c) {
+    try {
+      if (HEX_VALUES[c] < 0) {
+        throw new IllegalArgumentException("Illegal hexadecimal digit: " + c);
+      }
+    } catch (final ArrayIndexOutOfBoundsException e) {
+      throw new IllegalArgumentException("Illegal hexadecimal digit: " + c);
+    }
+
+    return HEX_VALUES[c];
+  }
+
+  public static long randomHalfLengthUUID() {
+    Random random = SentryUUID.Holder.numberGenerator;
+    byte[] randomBytes = new byte[8];
+    random.nextBytes(randomBytes);
+    randomBytes[6] &= 0x0f; /* clear version        */
+    randomBytes[6] |= 0x40; /* set to version 4     */
+
+    long msb = 0;
+
+    for (int i = 0; i < 8; i++) msb = (msb << 8) | (randomBytes[i] & 0xff);
+
+    return msb;
+  }
+
+  public static UUID randomUUID() {
+    Random random = SentryUUID.Holder.numberGenerator;
+    byte[] randomBytes = new byte[16];
+    random.nextBytes(randomBytes);
+    randomBytes[6] &= 0x0f; /* clear version        */
+    randomBytes[6] |= 0x40; /* set to version 4     */
+    randomBytes[8] &= 0x3f; /* clear variant        */
+    randomBytes[8] |= (byte) 0x80; /* set to IETF variant  */
```

Comment:
This method is taken from `java.uitl.UUID`.
Do we need this part? Do we want/need to add those flags to our random ids?